lib/bundler/injector.rb

```diff
@@ -90,5 +115,147 @@ def append_to(gemfile_path, new_gem_lines)
         f.puts new_gem_lines
       end
     end
+
+    # evalutes a gemfile to remove the specified gem
+    # from it.
+    def remove_deps(gemfile_path)
+      # get initial snap shot of the gemfile
+      initial_gemfile = IO.readlines(gemfile_path)
+
+      # inform user of the the gemfile currently
+      # being evaluated
+      Bundler.ui.info "Removing gems from #{gemfile_path}"
+
+      # evaluate the Gemfile we have
+      builder = Dsl.new
+      builder.eval_gemfile(gemfile_path)
+
+      # remove gems from dependencies
+      removed_deps = remove_gems_from_dependencies(builder, @deps)
+
+      # abort the opertion if no gems were removed
+      # no need to operate on gemfile furthur
+      return [] if removed_deps.empty?
+
+      # gemfile after removing requested gems
+      cleaned_gemfile = remove_gems_from_gemfile(@deps, gemfile_path)
+
+      # write the new gemfile
+      write_to_gemfile(gemfile_path, cleaned_gemfile)
+
+      # check for errors
+      # including extra gems being removed
+      # or some gems not being removed
+      # and return the actual removed deps
+      cross_check_for_errors(gemfile_path, builder.dependencies, removed_deps, initial_gemfile)
+    end
+
+    # @param [Dsl]    builder Dsl object of current Gemfile.
+    # @param [Array]  gems Array of names of gems to be removed.
+    # @return [Array] removed_deps Array of removed dependencies.
+    def remove_gems_from_dependencies(builder, gems)
+      removed_deps = []
+
+      gems.each do |gem_name|
+        deleted_dep = builder.dependencies.find {|d| d.name == gem_name }
+
+        if deleted_dep.nil?
+          raise GemfileError, "`#{gem_name}` is not specified in Gemfile so could not be removed."
```

Comment:
should use the actual path in the error message, and `so it could not be removed`